src/backend/dx11/src/device.rs

```diff
@@ -405,36 +488,44 @@ impl hal::Device<Backend> for Device {
         mem_type: hal::MemoryTypeId,
         size: u64,
     ) -> Result<Memory, device::OutOfMemory> {
-        let working_buffer_size = 1 << 15;
-        let working_buffer = if mem_type.0 == 1 {
-            let desc = d3d11::D3D11_BUFFER_DESC {
-                ByteWidth: working_buffer_size,
-                Usage: d3d11::D3D11_USAGE_STAGING,
-                BindFlags: 0,
-                CPUAccessFlags: d3d11::D3D11_CPU_ACCESS_READ | d3d11::D3D11_CPU_ACCESS_WRITE,
-                MiscFlags:0,
-                StructureByteStride: 0,
+        let heap_ty = MemoryHeapType::from_raw(mem_type.0).expect("invalid memory type id");
 
-            };
-            let mut working_buffer = ptr::null_mut();
-            let hr = unsafe {
-                self.raw.CreateBuffer(
-                    &desc,
-                    ptr::null_mut(),
-                    &mut working_buffer as *mut *mut _ as *mut *mut _
-                )
-            };
+        let (working_buffer, working_buffer_size) = match heap_ty {
+            MemoryHeapType::DEVICE_LOCAL => {
+                (None, 0)
+            },
+            MemoryHeapType::HOST_NONCOHERENT | MemoryHeapType::HOST_COHERENT => {
```

Comment:
this works as a pattern? O_O

Reply:
Yeah, left side only allows patterns, so eg. `1 + 2 => { ... }` wont compile

Reply:
oh, I get it now, this is not a match over the bitflag combo, this is a match over one of the two flags